frame/treasury/src/lib.rs

```diff
@@ -229,6 +231,11 @@ pub mod pallet {
 	pub type Approvals<T: Config<I>, I: 'static = ()> =
 		StorageValue<_, BoundedVec<ProposalIndex, T::MaxApprovals>, ValueQuery>;
 
+	/// The amount which has been reported as inactive to Currency.
+	#[pallet::storage]
+	pub type Deactivated<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, BalanceOf<T>, ValueQuery>;
```

Comment:
I see in https://github.com/paritytech/polkadot/pull/6372 you only `{re, de}activate` when funds are added to or withdrawn from a crowdloan. I suppose a storage item here is more sensible since with tx fees, era payouts, and spends, funds are more-or-less constantly going in and out of the Treasury via many avenues, and it'd be more error prone to capture the activation in every "to/from Treasury" hook?

Reply:
Precisely.